lms/djangoapps/bulk_email/models.py

```diff
@@ -70,7 +70,21 @@ class Meta(object):
         app_label = "bulk_email"
 
     def __unicode__(self):
-        return "CourseEmail Target for: {}".format(self.target_type)
+        return "CourseEmail Target: {}".format(self.short_display())
+
+    def short_display(self):
+        # django polymorphism is very odd, and doesn't work automatically
```

Comment:
django-model-utils contains an [InheritanceManager](https://django-model-utils.readthedocs.io/en/latest/managers.html#inheritancemanager) that handles model polymorphism pretty well.  A former coworker of mine wrote the code for it.  It could be worth a look.